lsp/nls/src/server.rs

```diff
@@ -107,8 +111,31 @@ impl Server {
         ));
     }
 
+    fn linearize_stdlib(&mut self) -> Result<()> {
+        self.cache.load_stdlib().unwrap();
+        let cache = &mut self.cache;
+        let modules = [
+            StdlibModule::Builtin,
+            StdlibModule::Contract,
+            StdlibModule::Array,
+            StdlibModule::Record,
+            StdlibModule::String,
+            StdlibModule::Num,
+            StdlibModule::Function,
+            StdlibModule::Internals,
+        ];
+        for module in modules {
+            let file_id = cache.get_submodule_file_id(module).unwrap();
+            cache
+                .typecheck_with_analysis(file_id, &self.initial_ctxt, &mut self.lin_cache)
+                .unwrap();
+        }
+        Ok(())
+    }
```

Comment:
This should use the `modules` function from `stdlib.rs` so we have a single source of truth on which modules are in the stdlib.